pkg/scheduler/scheduler.go

```diff
@@ -347,11 +357,13 @@ func updatePod(client clientset.Interface, pod *v1.Pod, condition *v1.PodConditi
 	podCopy := pod.DeepCopy()
 	// NominatedNodeName is updated only if we are trying to set it, and the value is
 	// different from the existing one.
-	if !podutil.UpdatePodCondition(&podCopy.Status, condition) &&
-		(len(nominatedNode) == 0 || pod.Status.NominatedNodeName == nominatedNode) {
+	if !podutil.UpdatePodCondition(&podCopy.Status, condition) && (pod.Status.NominatedNodeName == nominatedNode) {
 		return nil
 	}
-	if nominatedNode != "" {
+	// Update NominatedNodeName when the nominatedNode has a different value, the case nominatedNode is
+	// empty while the pod has NominatedNodeName set means the NominatedNodeName no longer fit,
+	// make it empty will moves the pod to the active queue. It lets scheduler find another place for the pod.
+	if podCopy.Status.NominatedNodeName != nominatedNode {
```

Comment:
feel like this fixes the issue #85677 also

Reply:
this change is necessary, actually, I found a race condition which is partially addressed by this change. I will give details about the race condition in my next day.

Reply:
pls see: https://github.com/kubernetes/kubernetes/issues/93505 for the race condition.